core/lib/zksync_core/src/api_server/web3/namespaces/en.rs

```diff
@@ -15,12 +15,33 @@ impl EnNamespace {
         Self { state }
     }
 
+    pub async fn consensus_genesis_impl(&self) -> Result<Option<en::ConsensusGenesis>, Web3Error> {
+        async {
```

Comment:
That's neat, but will be obsoleted by #1203, which removes `internal_error` altogether 🙃 Also, not very maintainable, because if you'd need to add a non-internal error response somewhere in the `async` block, it'd be silently converted to an internal error.

Reply:
outdated